lib/datadog/tracing/contrib/utils/quantization/http.rb

```diff
@@ -105,6 +107,21 @@ def collect_query(query, options = {})
             end
 
             private_class_method :collect_query
+
+            # Scans over the query string and obfuscates sensitive data by
+            # replacing matches with an opaque value
+            def obfuscate_query(query, options = {})
+              options[:regex] = nil if options[:regex] == :internal
+              re = options[:regex] || OBFUSCATOR_REGEX
+              with = options[:with] || OBFUSCATOR_WITH
+
+              query.gsub(re, with)
+            end
+
+            private_class_method :obfuscate_query
+
+            OBFUSCATOR_WITH = '<redacted>'.freeze
+            OBFUSCATOR_REGEX = /(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|token|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)(?:(?:\s|%20)*(?:=|%3D)[^&]+|(?:"|%22)(?:\s|%20)*(?::|%3A)(?:\s|%20)*(?:"|%22)(?:%2[^2]|%[^2]|[^"%])+(?:"|%22))|bearer(?:\s|%20)+[a-z0-9\._\-]+|token(?::|%3A)[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L](?:[\w=-]|%3D)+\.ey[I-L](?:[\w=-]|%3D)+(?:\.(?:[\w.+\/=-]|%3D|%2F|%2B)+)?|[\-]{5}BEGIN(?:[a-z\s]|%20)+PRIVATE(?:\s|%20)KEY[\-]{5}[^\-]+[\-]{5}END(?:[a-z\s]|%20)+PRIVATE(?:\s|%20)KEY|ssh-rsa(?:\s|%20)*(?:[a-z0-9\/\.+]|%2F|%5C|%2B){100,}/
```

Comment:
I put these constants at the end because when at top the complex regex throws off syntax highlighters and makes editing a pain.

Reply:
Can I convince you to format this regular expression a bit?
Some like [`%r{}x` for multiline regex](https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#regular-expressions) is probably ok.

Also, given that every large enough regex is virtually unintelligible, could you write a small comment explaining, at a high level, what this regex tries to match?

Reply:
Definitely. I considered it as well, glad we're on the same mind!

That said, including that regex is kind of awkward since it's obtained from an internal document which might be updated, and should it be updated it would require re-formatting it every time, at the risk of breaking it.

Reply:
> That said, including that regex is kind of awkward since it's obtained from an internal document which might be updated, and should it be updated it would require re-formatting it every time, at the risk of breaking it.

Then don't touch it :) Let's not make our lives harder. Documentation is enough then.

Reply:
I needed to improve on it and fix some corner cases, so I guess I'll update it in the doc with the multiline version.